spieleplattform-frontend/src/components/Game/AddGame/AddGame.tsx

```diff
@@ -0,0 +1,125 @@
+import "./AddGame.scss"
+import "react-datepicker/dist/react-datepicker.min.css";
+import DatePicker from "react-datepicker";
+import { useEffect, useState } from "react";
+import GameConsole from "../GameConsole";
+import axios from "axios";
+import { useKeycloak } from "@react-keycloak/web";
+import Error from "../../Error/Error";
+import AddGameDTO from "../AddGameDTO";
+import Game from "../Game";
+
+export default function AddGame() {
+    const [gameConsoles, setGameConsoles] = useState<GameConsole[] | null>(null)
+    const [gameConsoleSelectedFlags, setGameConsoleSelectedFlags] = useState<Boolean[]>(new Array(0))
+    const [errorStatus, setErrorStatus] = useState(null)
+    const { keycloak } = useKeycloak()
+
+    const [name, setName] = useState<string>("")
+    const [releaseDate, setReleaseDate] = useState<Date>(new Date())
+    const [imageUrl, setImageUrl] = useState<string>("")
+    const [youtubeId, setYoutubeId] = useState<string>("")
+    const [description, setDescription] = useState<string>("")
+
+    useEffect(() => {
+        axios.get("/gameConsoles", { headers: { 'Authorization': `Bearer ${keycloak.token}` } })
+            .then((response) => {
+                setGameConsoles(response.data)
+                setGameConsoleSelectedFlags(Array(response.data.length).fill(false))
+            })
+            .catch(error => {
+                setErrorStatus(error.response.status)
+            })
+    }, [keycloak.token])
+
+    function toggleGameConsoleSelectedFlag(index: number) {
+        if (gameConsoleSelectedFlags == null) return
+
+        gameConsoleSelectedFlags[index] = !gameConsoleSelectedFlags[index]
+        setGameConsoleSelectedFlags(structuredClone(gameConsoleSelectedFlags))
+    }
+
+    function submitGame() {
+        if (gameConsoles == null) return
+
+        const gameConsoleIds = new Array<number>(0);
+        for(let index = 0; index < gameConsoles?.length; index++) {
+            if (gameConsoleSelectedFlags[index]) {
+                gameConsoleIds.push(gameConsoles[index].id)
```

Comment:
Auch hier wäre es wieder cool wenn du um den for-loop drum rum kommst. Ich glaube du machst es dir hier aber auch komplizierter als es sein müsste. Anstelle mit Indexen und einem boolschen Array zu arbeiten, könntest du zum Beispiel auch ein Array haben mit den Konsolen die gerade selektiert sind. Oder du könntest dir auch einen Typ definieren der dein GameConsole-Objekt um eine isSelected Proberty erweitert. Können wir uns auch gerne am Montag mal gemeinsam anschauen, wenn du willst. :)